### PR TITLE
add_pangwes

### PR DESCRIPTION
**Pull request by Sudaraka88**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/pangwes/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# build and install PAN-GWES
+make -C sw/pangwes
```
Comment (martin-g) [inline recipes/pangwes/build.sh:4]:
```suggestion
make -C sw/pangwes -j ${CPU_COUNT}
```

File: recipes/pangwes/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
```
Comment (martin-g) [inline recipes/pangwes/build.sh:2]:
```suggestion

set -xe

```

File: recipes/pangwes/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+# build and install PAN-GWES
+make -C sw/pangwes
+pwd
```
Comment (martin-g) [inline recipes/pangwes/build.sh:5]:
```suggestion

```

File: recipes/pangwes/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/Sudaraka88/PAN-GWES/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: e72ec296e7b79d5ecd7645b5db96983600b2093d556ca2f4954e45197225eb46
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name, max_pin="x") }}
```
Comment (martin-g) [inline recipes/pangwes/meta.yaml:15]:
```suggestion
    - {{ pin_subpackage(name, max_pin="x.x") }}
```

File: recipes/pangwes/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+requirements:
+  script:
+    - export CC=$CONDA_PREFIX/bin/g++  # Set the compiler path
+  build:
+    - gxx
```
Comment (martin-g) [inline recipes/pangwes/meta.yaml:21]:
```suggestion
    - {{ compiler('cxx') }}
```

File: recipes/pangwes/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  sha256: e72ec296e7b79d5ecd7645b5db96983600b2093d556ca2f4954e45197225eb46
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name, max_pin="x") }}
+
+requirements:
+  script:
+    - export CC=$CONDA_PREFIX/bin/g++  # Set the compiler path
```
Comment (martin-g) [inline recipes/pangwes/meta.yaml:19]:
```suggestion
```

File: recipes/pangwes/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://github.com/Sudaraka88/PAN-GWES/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: e72ec296e7b79d5ecd7645b5db96983600b2093d556ca2f4954e45197225eb46
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name, max_pin="x") }}
+
+requirements:
+  script:
```
Comment (martin-g) [inline recipes/pangwes/meta.yaml:18]:
```suggestion
```

File: recipes/pangwes/build.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -xe
+
+# build and install PAN-GWES
+make -C sw/pangwes -j ${CPU_COUNT}
```
Comment (martin-g) [inline recipes/pangwes/build.sh:6]:
```suggestion
make -C sw/pangwes -j ${CPU_COUNT}
find ./ -name "*unitig_distance*"
```

File: recipes/pangwes/build.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+find ./ -name "*unitig_distance*"
+cp $SRC_DIR/sw/pangwes/bin/unitig_distance $PREFIX/bin/unitig_distance
```
Comment (martin-g) [inline recipes/pangwes/build.sh:8]:
```suggestion
cp ./sw/pangwes/bin/unitig_distance $PREFIX/bin/unitig_distance
```

File: recipes/pangwes/build.sh [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+cp $SRC_DIR/sw/pangwes/bin/gfa1_parser $PREFIX/bin/gfa1_parser
```
Comment (martin-g) [inline recipes/pangwes/build.sh:9]:
```suggestion
cp ./sw/pangwes/bin/gfa1_parser $PREFIX/bin/gfa1_parser
```

File: recipes/pangwes/build.sh [hunk 11]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -xe
+
+# build and install PAN-GWES
+make -C sw/pangwes -j ${CPU_COUNT}
+find ./ -name "*unitig_distance*"
```
Comment (martin-g) [inline recipes/pangwes/build.sh:7]:
```suggestion
mkdir -p ${PREFIX}/bin
```

**Comment by Sudaraka88** [comment]
Hello @martin-g, thank you very much for your help with fixing the meta/build files. This is my first time submitting a C++ package to bioconda...

I'm not the original author of this package and I could only compile it using g++, which is why I had this bit in the meta file: 
`export CC=$CONDA_PREFIX/bin/g++ # Set the compiler path`
I think it builds with g++ on both linux64 and osx-64

Currently, the macOS build is failing with the same issue I had locally when using gcc:
`2024-06-20T11:23:44.8311170Z 11:23:40 32m BIOCONDA INFO 0m (ERR) include/unitig_distance/Queries.hpp:35:40: error: no member named 'emplace_back' in 'std::vector<bool>'[0m`

Just wondering if the make command needs to be changed or some other flags are needed for osx? 
Thanks again!

**Pull request closed without merging**